pkg/cmd/roachtest/registry/filter.go

```diff
@@ -11,14 +11,30 @@
 package registry
 
 import (
+	"fmt"
 	"regexp"
+	"sort"
 	"strings"
+	"unicode"
 )
 
 // TestFilter holds the name and tag filters for filtering tests.
 // See NewTestFilter.
 type TestFilter struct {
```

Comment:
Since the concept of `SuiteSet` and `CloudSet` now already exist, why not here here too? It would allow users to specify tests that belong to multiple suites,  or owners, or clouds (less useful).